pkg/dns/ibm/public/cis_provider.go

```diff
@@ -103,7 +101,11 @@ func (p *Provider) Delete(record *iov1.DNSRecord, zone configv1.DNSZone) error {
 	}
 	opt := dnsService.NewListAllDnsRecordsOptions()
 	opt.SetType(string(record.Spec.RecordType))
-	opt.SetName(record.Spec.DNSName)
+	// DNS records may have an ending "." character in the DNS name.  For
+	// example, the ingress operator's ingress controller adds a trailing
+	// "." when it creates a wildcard DNS record.
+	dnsName := strings.TrimSuffix(record.Spec.DNSName, ".")
+	opt.SetName(dnsName)
```

Comment:
If the record was created with a trailing dot, on the CIS side it still appears without the dot. This means ListRecords won't return the records as it will look for records with a trailing dot. It was fixed for update/create here: https://github.com/openshift/cluster-ingress-operator/pull/652, but I believe it should have been added to delete as well.

Reply:
Does that mean that your change here fixes an unreported defect where the operator silently failed to delete old DNS records (since it would look for a record with a trailing ".", fail to find it, and assume there was nothing to delete)?  If so, nice!  And we might want to call that out in a release note.